packages/backend/src/peripherals/database/shared/types.ts

```diff
@@ -155,6 +153,59 @@ declare module 'knex/types/tables' {
     cancelled_at: Nullable<bigint>
   }
 
+  interface ForcedWithdrawTransactionRow {
+    hash: string
+    stark_key: string
+    amount: bigint
+    position_id: bigint
+  }
+
+  interface ForcedWithdrawStatusRow {
+    hash: string
+    status: 'sent' | 'forgotten' | 'reverted' | 'mined' | 'included'
+    timestamp: bigint
+    block_number: Nullable<number>
+    state_update_id: Nullable<number>
+  }
+
+  interface ForcedTradeTransactionRow {
+    hash: string
+    stark_key_a: string
+    stark_key_b: string
+    position_id_a: bigint
+    position_id_b: bigint
+    collateral_amount: bigint
+    synthetic_amount: bigint
+    is_a_buying_synthetic: boolean
+    synthetic_asset_id: string
+    nonce: bigint
+  }
+
+  interface ForcedTradeStatusRow {
+    hash: string
+    status: string
+    timestamp: bigint
+    block_number: Nullable<number>
+    offer_id: Nullable<number>
+    state_update_id: Nullable<number>
+  }
+
+  interface WithdrawTransactionRow {
+    hash: string
+    stark_key: string
+    asset_type: string
+  }
+
+  interface WithdrawStatusRow {
+    hash: string
+    status: string
+    timestamp: bigint
+    block_number: Nullable<number>
+    quantized_amount: Nullable<bigint>
+    non_quantized_amount: Nullable<bigint>
+    recipient_address: Nullable<string>
+  }
+
```

Comment:
Do you think it's better to have Statuses types in `ForcedWithdrawRepository` instead of here?

Reply:
The reason the statuses (but only names) are also declared in the types is to make sure that the `switch` at https://github.com/l2beat/starkex-explorer/pull/245/files#diff-5cac483fadd9b48a926df5ff307622a120daac55611bcf9f5286ed9f172f3f1aR324 is exhaustive.

Reply:
Ok, that brings me a different thing I now thought of. Would it be better to have an enum for the statuses names?

Reply:
Enums have a lot of problems in TypeScript, because they work differently than other language features. Most of what TypeScript does happens only on the type level and gets removed during compilation. Enums persist and stay in the code, which is unintuitive and makes them not play nicely with other typescript constructions that don't behave in this way. Enums are also hard to debug because they are essentially variables with primitive values (usually consecutive integers) assigned to them, so if you console log an enum you don't get any indication that this value is an enum, it will just be a regular primitive